src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+        helperTextInvalid={t("common:required")}
+      >
+        <Controller
+          name="config.principalType"
+          defaultValue={t("subjectNameId")}
+          control={control}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-principalType"
+              onToggle={() =>
+                setPrincipalTypeDropdownOpen(!principalTypeDropdownOpen)
+              }
+              isOpen={principalTypeDropdownOpen}
+              onSelect={(_, value) => {
+                onChange(value as string);
+                setPrincipalTypeDropdownOpen(false);
+              }}
+              selections={value}
+              variant={SelectVariant.single}
+            >
+              <SelectOption
+                key={0}
+                data-testid="subjectNameId-option"
+                value={t("subjectNameId")}
+                isPlaceholder
+              />
+              <SelectOption
+                key={1}
+                data-testid="attributeName-option"
+                value={t("attributeName")}
+              />
+              <SelectOption
+                key={2}
+                data-testid="attributeFriendlyName-option"
+                value={t("attributeFriendlyName")}
+              />
+            </Select>
+          )}
+        ></Controller>
+      </FormGroup>
+
+      <SwitchField
+        field="config.postBindingResponse"
+        label="httpPostBindingResponse"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.postBindingAuthnRequest"
+        label="httpPostBindingAuthnRequest"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.postBindingLogout"
+        label="httpPostBindingLogout"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.wantAuthnRequestsSigned"
+        label="wantAuthnRequestsSigned"
+        isReadOnly={readOnly}
+      />
+
+      {wantAuthnSigned === "true" && (
+        <>
+          <FormGroup
+            label={t("signatureAlgorithm")}
+            labelIcon={
+              <HelpItem
+                helpText={th("signatureAlgorithm")}
+                forLabel={t("signatureAlgorithm")}
+                forID="kc-signatureAlgorithm"
+              />
+            }
+            fieldId="kc-signatureAlgorithm"
+          >
+            <Controller
+              name="config.signatureAlgorithm"
+              defaultValue="RSA_SHA256"
+              control={control}
+              render={({ onChange, value }) => (
+                <Select
+                  toggleId="kc-signatureAlgorithm"
+                  onToggle={() =>
+                    setSignatureAlgorithmDropdownOpen(
+                      !signatureAlgorithmDropdownOpen
+                    )
+                  }
+                  isOpen={signatureAlgorithmDropdownOpen}
+                  onSelect={(_, value) => {
+                    onChange(value as string);
```

Comment:
There is no need to cast this type as `SelectOptionObject` implements `toString()`.
```suggestion
                    onChange(value.toString());
```

Reply:
yep, all these were fixed last iteration, good catches